### ISSUE TITLE
Issue with ddp + gradient checkpointing

### ISSUE THREAD

**Issue by jw3126** [PyTorchLightning/pytorch-lightning#4788]
## 🐛 Bug
From [the forum](https://forums.pytorchlightning.ai/t/gradient-checkpointing-ddp-nan/398/). The [model](https://gist.github.com/jw3126/29cbb8177a9f4acfb66fc3a3ed7cbf24) runs fine on one gpu, but loss becomes `nan` on multiple gpus.
This seems to be an issue when combining gradient checkpointing and ddp.

### To Reproduce

Run the following with `gpus=1` and `gpus=4`
https://gist.github.com/jw3126/29cbb8177a9f4acfb66fc3a3ed7cbf24

### Expected behavior

Loss should be finite independently of `gpus=1` or `gpus=4`

### Environment
```
* CUDA:
        - GPU:
                - Tesla K80
                - Tesla K80
                - Tesla K80
                - Tesla K80
        - available:         True
        - version:           10.1
* Packages:
        - numpy:             1.18.5
        - pyTorch_debug:     True
        - pyTorch_version:   1.7.0+cu101
        - pytorch-lightning: 1.0.5
        - tqdm:              4.51.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.6.9
        - version:           #32~18.04.1-Ubuntu SMP Tue Oct 6 10:03:22 UTC 2020
```

**Comment by robogast** [comment]
I've been able to reproduce your results; with `gpus=1` it trains stabily, and with `gpus=4` it collapses to `nan`.

I don't know anything about gradient checkpointing, but some time ago I was able to fix my own instability issues by setting `amsgrad=True` or `eps=1e-2` for Adam: I tested this on your code but that didn't help.

Just putting an egg into the basket :slightly_smiling_face:

**Comment by timothybrooks** [comment]
I too would love to see gradient checkpointing working with lightning and ddp. FairScale seems to use gradient checkpointing with ddp in their tests successfully, so it should be possible.

https://github.com/facebookresearch/fairscale/blob/66b2b5145ef9799f64e8056858c43609ec936bea/tests/nn/pipe/test_checkpoint_ddp.py

**Comment by timothybrooks** [comment]
I was able to use DDP and Pytorch gradient checkpointing using the following:

```
ddp = pytorch_lightning.plugins.ddp_plugin.DDPPlugin(find_unused_parameters=False)
trainer = pytorch_lightning.Trainer()
trainer.fit(model, plugins=[ddp])
```
